Rewrite diff check to match example in cmp.Diff docs

mlevesquedion opened the pull request:
Fixes #4664.

puradox reviewed (approved):
Thank you for doing this! 🙌

mlevesquedion commented:
Of course! Happy to lend a hand :)

avagin commented:
@mlevesquedion sorry for the delay. Could you rebase the PR?

mlevesquedion commented:
No worries. I've rebased the PR.

mlevesquedion commented:
There was another merge conflict. I've rebased the PR again.

tamird commented on pkg/tcpip/stack/neighbor_cache_test.go:
```diff
@@ -1750,6 +1750,34 @@ func TestNeighborCacheRetryResolution(t *testing.T) {
 	}
 }
 
+// TestNeighborCacheStaticResolution checks that static link addresses are
```
looks like bad conflict resolution. this test is failing.

mlevesquedion replied:
Sorry about that. Should be fixed now.

avagin commented:
```
$ bazel test pkg/tcpip/stack:stack_test
INFO: Analyzed target //pkg/tcpip/stack:stack_test (96 packages loaded, 7852 targets configured).
INFO: Found 1 test target...
FAIL: //pkg/tcpip/stack:stack_test (see bazel-out/k8-fastbuild/testlogs/pkg/tcpip/stack/stack_test/test.log)
Target //pkg/tcpip/stack:stack_test up-to-date:
  bazel-bin/pkg/tcpip/stack/stack_test_/stack_test
INFO: Elapsed time: 35.493s, Critical Path: 16.57s
INFO: 110 processes: 4 internal, 106 linux-sandbox.
INFO: Build completed, 1 test FAILED, 110 total actions
//pkg/tcpip/stack:stack_test                                             FAILED in 11.9s
  bazel-out/k8-fastbuild/testlogs/pkg/tcpip/stack/stack_test/test.log

INFO: Build completed, 1 test FAILED, 110 total actions
$ cat bazel-out/k8-fastbuild/testlogs/pkg/tcpip/stack/stack_test/test.log
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //pkg/tcpip/stack:stack_test
-----------------------------------------------------------------------------
--- FAIL: TestNeighborCacheStaticResolution (0.00s)
    neighbor_cache_test.go:1769: unexpected error from neigh.entry(62726f616463617374, '', _, nil, nil): operation would block
FAIL
```

mlevesquedion commented:
Sorry about that. The failing test should not be there, I have removed it. Also there were some unintended changes to `go.{mod,sum}` that I have just removed.
